library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.ts

```diff
@@ -224,25 +221,21 @@ export class BankAccountConnectComponent implements OnInit {
         .pipe(
           take(1),
           switchMap((bank) => {
-            // Default asset to 'USD' for non-production banks
-            const asset =
-              bank.type == BankBankModel.TypeEnum.Sandbox
-                ? bank.supported_fiat_account_assets![0]
-                : account.iso_currency_code;
-
-            console.log(this.config.environment);
-            console.log(asset);
-            console.log(metadata.accounts);
-            console.log(bank.supported_fiat_account_assets!.includes(asset));
-
-            if (bank.supported_fiat_account_assets!.includes(asset)) {
+            if (
+              bank.supported_fiat_account_assets!.includes(
+                account.iso_currency_code
+              ) ||
+              account.iso_currency_code == undefined
+            ) {
               return this.bankAccountService.createExternalBankAccount(
                 account.name,
                 public_token,
                 account.id,
-                asset
+                account.iso_currency_code
               );
-            } else return throwError(() => new Error('Unsupported asset'));
+            } else {
+              return throwError(() => new Error('Unsupported asset'));
```

Comment:
Is the user notified there's a problem?

This is an interim solution. We need to add the prompt to the user for them to choose based on the bank's config (follow-on PR)

Reply:
The error is caught and the ui returns an undefined error.

Copy. It shouldn't be difficult to add a modal with a select for assets.